pkg/controller/authentication/authentication_controller.go

```diff
@@ -0,0 +1,269 @@
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, provider oprv1.Provider) *ReconcileAuthentication {
+	r := &ReconcileAuthentication{
+		client:   mgr.GetClient(),
+		scheme:   mgr.GetScheme(),
+		provider: provider,
+		status:   status.New(mgr.GetClient(), "authentication"),
+	}
+	r.status.Run()
+	return r
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r *ReconcileAuthentication) error {
+	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return fmt.Errorf("failed to create %s: %w", ControllerName, err)
+	}
+
+	err = c.Watch(&source.Kind{Type: &oprv1.Authentication{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("%s failed to watch resource: %w", ControllerName, err)
+	}
+
+	err = c.Watch(&source.Kind{Type: &oprv1.ManagementClusterConnection{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("%s failed to watch resource: %w", ControllerName, err)
+	}
+
+	for _, namespace := range []string{render.OperatorNamespace(), render.DexNamespace} {
+		for _, secretName := range []string{
+			render.DexTLSSecretName, render.OIDCSecretName, render.OpenshiftSecretName, render.DexObjectName,
+		} {
+			if err = utils.AddSecretsWatch(c, secretName, namespace); err != nil {
+				return fmt.Errorf("%s failed to watch the secret '%s' in '%s' namespace: %w", ControllerName, secretName, namespace, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileAuthentication implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileAuthentication{}
+
+// ReconcileAuthentication reconciles an Authentication object
+type ReconcileAuthentication struct {
+	client   client.Client
+	scheme   *runtime.Scheme
+	provider oprv1.Provider
+	status   status.StatusManager
+}
+
+// Reconciles the cluster state with the Authentication object that is found in the cluster.
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileAuthentication) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ", "controller", ControllerName)
+
+	ctx := context.Background()
+
+	// Fetch the Authentication spec. If present, we deploy dex in the cluster
+	authentication, err := utils.GetAuthentication(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.OnCRNotFound()
+		}
+		return reconcile.Result{}, err
+	}
+	r.status.OnCRFound()
+	reqLogger.V(2).Info("Loaded config", "config", authentication)
+
+	// Query for the installation object.
+	network, err := installation.GetInstallation(context.Background(), r.client)
```

Comment:
This looks like a copy/paste from some old code from before the Installation object was called "Installation" (it used to be called network!)

I'd s/network/installation here to make the names agree.

Reply:
I thought we did that because installation collides with the package name.

Reply:
Ya it would be great to not use a name that conflicts, maybe we can just change them all in another PR. You can't use anything from the installation package after this point if network was changed to installation, I wonder if we can enable a linter or something to catch this.